### PR TITLE
Issue 4591 - Custom ticket groups - UX Improvements

### LINKED ISSUE #4591: Custom ticket groups - UX improvements
### Description
A few UX weirdness that can be addressed around custom ticket groups

### Goals
- [x] When the user tries to hide all hidden groups, it should deselect any selected groups within the category
img-13-14-23-03-14

- [x] Newly created hidden groups should be hidden by default instead of shown.
img-13-15-23-03-15

- [x] If the group you have selected is deleted, it should not auto select the next group
img-13-19-23-03-19

### PR DESCRIPTION
**Pull request by Amantini1997**
This fixes #4591

#### Description
- When the user tries to hide all hidden groups, it should deselect any selected groups within the category
- Newly created hidden groups should be hidden by default instead of shown
- If the group you have selected is deleted, it should not auto select the next group

#### Test cases
Test them

### CODE REVIEW

**Comment by The-Daniel** [comment]
Another few problems that could possibly be addressed???
1 - Should clicking 'Show all' in the toolbar change the visibility in the groups list? (Note the eye icon is not changing)
image
image

2 - If you have a group list item selected and then click 'Clear overrides' in the toolbar the override briefly flickers off but stays highlighted. Should this deselect the list item and remove the override?

(Sorry. I know these will require diving into v4 code😬)

File: frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/ticketGroups/groups/groupItem/groupItem.component.tsx [hunk 1]
```diff
@@ -60,6 +60,8 @@ export const GroupItem = ({ override, index }: GroupProps) => {
 	const isAdmin = ProjectsHooksSelectors.selectIsProjectAdmin();
 	const { group, color, opacity } = override;
 	const checked = getGroupIsChecked(index);
+	const isHidden = groupType === 'hidden';
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/ticketGroups/groups/groupItem/groupItem.component.tsx:63]: When editing an existing hidden group it the group stays selected showing the same problem
show hidden group

File: frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/ticketGroups/groups/groupItem/groupItem.component.tsx [hunk 2]
```diff
@@ -101,27 +100,39 @@ export const GroupItem = ({ override, index }: GroupProps) => {
...
+			highlightGroupObjects();
+		}
+	}, [override, setHighlightedIndex]);
+
+	useEffect(() => {
+		if (!isHighlighted && isHidden && checked) {
+			const objects = convertToV4GroupNodes((group as Group).objects);
+			dispatch(TreeActions.hideNodesBySharedIds(objects));
+		}
+	}, [isHighlighted]);
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/ticketGroups/groups/groupItem/groupItem.component.tsx:130]: what is this new code for?

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

**Comment by sanmont3drepo** [comment]
> Another few problems that could possibly be addressed??? 1 - Should clicking 'Show all' in the toolbar change the visibility in the groups list? (Note the eye icon is not changing) image image
> 
> 2 - If you have a group list item selected and then click 'Clear overrides' in the toolbar the override briefly flickers off but stays highlighted. Should this deselect the list item and remove the override?
> 
> (Sorry. I know these will require diving into v4 code😬)

This will be addressed in #4617

**Review by sanmont3drepo — APPROVED** [review]

**Pull request merged**